src/services/azure/credential.rs

```diff
@@ -0,0 +1,128 @@
+use std::fmt::{Debug, Formatter};
+use std::mem;
+use std::ops::Add;
+use std::time::SystemTime;
+
+use anyhow::{anyhow, Result};
+use time::Duration;
+
+#[derive(Default)]
+pub struct Builder {
+    cred: Credential,
+}
+
+impl Builder {
+    pub fn access_acount(&mut self, access_acount: &str) -> &mut Self {
+        self.cred.access_acount = access_acount.to_string();
+        self
+    }
+    pub fn access_key(&mut self, access_key: &str) -> &mut Self {
+        self.cred.access_key = access_key.to_string();
+        self
+    }
+    pub fn sas_token(&mut self, sas_token: &str) -> &mut Self {
```

Comment:
It looks like `sas_token` was never used in signing, how about removing it now? It could be confused if the user tries to use this feature.

Reply:
sas_token is another way to sign in azure . I'm not sure it's necessary to implement this feature yet so I'm keeping it. I'll delete it first and plan later